src/openlifu/plan/solution.py

```diff
@@ -64,14 +102,221 @@ class Solution:
 
     approved: bool = False
     """Approval state of this solution as a sonication plan. `True` means the user has provided some
     kind of confirmation that the solution is safe and acceptable to be executed."""
 
-    def to_json(self, include_simulation_data:bool, compact:bool) -> str:
+    def num_foci(self):
+        """Get the number of foci"""
+        if isinstance(self.focus, list):
+            nfoc = len(self.focus)
+        elif isinstance(self.focus, Point):
+            nfoc = 1
+        else:
+            raise ValueError("Cannot get number of foci for types other than Point.")
+
+        return nfoc
+
+    def analyse(self, transducer: Transducer, options: SolutionOptions = SolutionOptions()) -> SolutionAnalysis:
+        """Analyzes the treatment solution.
+
+        Args:
+            transducer: A Transducer item.  #TODO: this should be instanciated at the database level, not here ?
+            options: A struct for solution analysis options.
+
+        Returns: A struct containing the results of the analysis.
+        """
+        solution_analysis = SolutionAnalysis()
+
+        if transducer.id != self.transducer_id:
+            # self.logger.error(f"Provided transducer id {transducer.id} does not match Solution transducer id: {self.transducer_id}")
+            raise ValueError(f"Provided transducer id {transducer.id} does not match Solution transducer id: {self.transducer_id}")
+
+        # dt = 1 / (self.pulse.frequency * 20)
+        # t = self.pulse.calc_time(dt)
+        # input_signal = self.pulse.calc_pulse(t)
+
+        pnp = self.simulation_result.p_min
+        ppp = self.simulation_result.p_max
+        ita = self.simulation_result.ita
+
+        if options.sidelobe_radius is np.nan:
+            options.sidelobe_radius = options.mainlobe_radius
+
+        pnp_MPa = rescale_data_arr(pnp, "MPa")
+
+        # standoff_Z = options.standoff_density * 1500
+        # c_tic = 40e-3  # W cm-1
+        # A_cm = transducer.get_area("cm")
+        # d_eq_cm = np.sqrt(4*A_cm / np.pi)
+        # ele_sizes_cm2 = np.array([elem.get_area("cm") for elem in transducer.elements])
+
+        # xyz = np.stack(np.meshgrid(*coords, indexing="xy"), axis=-1)  #TODO: if fus.Axis is defined, coords.ndgrid(dim="ax")
+        # z_mask = xyz[..., -1] >= options.sidelobe_zmin  #TODO: probably wrong here, should be z{1}>=options.sidelobe_zmin;
+
+        # intensity_Wcm2 = output.intensity.rescale_data("W/cm^2")
+        # pulsetrain_dutycycle = self.get_pulsetrain_dutycycle()
+        # treatment_dutycycle = self.get_treatment_dutycycle()
+        # ita_mWcm2 = self.get_ita(output, units="W/cm^2")
+
+        # power_W = np.zeros(self.num_foci())
+        # TIC = np.zeros(self.num_foci())
+        for focus_index in range(self.num_foci()):
+            if isinstance(self.focus, list):
+                foc = self.focus[focus_index]
+            elif isinstance(self.focus, Point):
+                foc = self.focus
```

Comment:
Since you are iterating over foci here already, I feel that we could merge #127 first before this so that this can rely on the list of foci always being an actual list. If you agree please go ahead and review #127

Reply:
I prefer to add the foci list feature later so we keep the minimal solution clean of this.

I added it because it was some leftovers from my initial development, but the rest is actually not really compatible with a foci list (few attributes from `solution_analysis` should be a list).